### PR TITLE
chore: migrate governance + market pages onto trading rewards page (flagged)

### PR DESCRIPTION
**Pull request by moo-onthelawn**
Screenshot 2024-05-30 at 10 31 54 AM

---

## Views

* `token/Token.tsx`
  * Hide (do not render) sidebar to navigate between Rewards/Staking/Governance when `tradingRewardsRehaul` flag is on

* `RewardsPage`
  * Update to render `Governance` and `NewMarkets` panels on the page behind `tradingRewardsRehaul` flag
    * Applies to tablet + web view
    * Left some placeholders for new UI that has yet to be built
  * Refactored how we specify when + what we render (in tablet vs web view) because it was so hard to parse  

## Components

* New: `RewardsNavPanel`
  * Extracted out common component from `GovernancePanel` and `NewMarketsPanel` since they share the same styling

* `GovernancePanel` / `NewMarketsPanel`
  * Updated to use `RewardsNavPanel`
  * Made small flagged edits to text + styling for when it's rendered on the token page (flagged behind `tradingRewardsRehaul`)
  * I could've made new components instead of flagging, but this seemed easy enough to read and clean up later 

* `RewardHistoryPanel`
  * Update text alignment on smaller screens to match figma [mock](https://www.figma.com/design/dhSd8kkcUrY3UBRsdS89kH/dYdX-%E2%80%BA-Legal?node-id=946-5907&m=dev)

* `LaunchIncentivesPanel`
  * Update styling of AboutButton and LeaderboardButton so that LeaderboardButton grows twice as fast (and fills the remaining space) to match [mock](https://www.figma.com/design/dhSd8kkcUrY3UBRsdS89kH/dYdX-%E2%80%BA-Legal?node-id=946-5773&m=dev)
   
Screenshot 2024-05-29 at 4 41 07 PM

## Packages

* `localization`
  * pull in updated legal text
  * updated: v1.1.92 -> v1.1.96

## Testing

* `?tradingRewardsRehaul=true`
- [x] Verify that sidebar still shows when flag is off
- [x] Verify governance panel looks normal when flag is off; verify clickthrough behavior
- [x] Verify new markets panel looks normal when flag is off; verify clickthrough behavior
- [x] Verify clickthrough behavior on governance panel when flag is on
- [x] Verify clickthrough behavior on new markets panel when flag is on
- [x] Verify tablet view when flag is off
- [x] Verify tablet view when flag is on

---

### CODE REVIEW

File: src/pages/token/rewards/RewardsNavPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            action={ButtonAction.Base}
+            iconName={IconName.Arrow}
+            size={ButtonSize.Small}
+          />
+        </$Arrow>
+      }
+      onClick={onNav}
+    >
+      <$Description>
+        <>
```
Comment (aforaleka) [inline src/pages/token/rewards/RewardsNavPanel.tsx:61]: do we need the `<></>`?

File: src/pages/token/rewards/RewardsPage.tsx [hunk 2]
```diff
@@ -29,111 +31,136 @@ const RewardsPage = () => {
...
-  const showChartPanel = testFlags.tradingRewardsRehaul;
+  const tradingRewardsRehaulEnabled = testFlags.tradingRewardsRehaul;
 
-  return (
+  const legalDisclaimer = (
+    <$LegalDisclaimer>
+      {stringGetter({ key: STRING_KEYS.TRADING_REWARDS_LEGAL_DISCLAIMER })}
+    </$LegalDisclaimer>
+  );
+  return isTablet ? (
```
Comment (aforaleka) [inline src/pages/token/rewards/RewardsPage.tsx:41]: nit: for mobile view, the table can now overflow horizontally whereas it couldn't previously, can we fix that in css back?
Screenshot 2024-05-31 at 1 33 17 PM

Comment (moo-onthelawn) [reply src/pages/token/rewards/RewardsPage.tsx:42]: yess fixed!
